airflow/models/trigger.py

```diff
@@ -122,17 +122,19 @@ def submit_event(cls, trigger_id, event, session=None):
         Takes an event from an instance of itself, and triggers all dependent
         tasks to resume.
         """
-        for task_instance in session.query(TaskInstance).filter(
-            TaskInstance.trigger_id == trigger_id, TaskInstance.state == State.DEFERRED
-        ):
-            # Add the event's payload into the kwargs for the task
-            next_kwargs = task_instance.next_kwargs or {}
-            next_kwargs["event"] = event.payload
-            task_instance.next_kwargs = next_kwargs
-            # Remove ourselves as its trigger
-            task_instance.trigger_id = None
-            # Finally, mark it as scheduled so it gets re-queued
-            task_instance.state = State.SCHEDULED
+        for attempt in run_with_db_retries():
+            with attempt:
+                for task_instance in session.query(TaskInstance).filter(
+                    TaskInstance.trigger_id == trigger_id, TaskInstance.state == State.DEFERRED
+                ):
+                    # Add the event's payload into the kwargs for the task
+                    next_kwargs = task_instance.next_kwargs or {}
+                    next_kwargs["event"] = event.payload
+                    task_instance.next_kwargs = next_kwargs
+                    # Remove ourselves as its trigger
+                    task_instance.trigger_id = None
+                    # Finally, mark it as scheduled so it gets re-queued
+                    task_instance.state = State.SCHEDULED
```

Comment:
```suggestion
        for attempt in run_with_db_retries():
            with attempt:
                tis = session.query(TaskInstance).filter(
                    TaskInstance.trigger_id == trigger_id, TaskInstance.state == State.DEFERRED
                )

        for task_instance in tis:
            # Add the event's payload into the kwargs for the task
            next_kwargs = task_instance.next_kwargs or {}
            next_kwargs["event"] = event.payload
            task_instance.next_kwargs = next_kwargs
            # Remove ourselves as its trigger
            task_instance.trigger_id = None
            # Finally, mark it as scheduled so it gets re-queued
            task_instance.state = State.SCHEDULED
```

i think separating the retry logic from the actual work makes things a little more digestable

Reply:
actually that might not work

Reply:
because we're mutating the objects and probably this actually fails when the method exits.

so, looking at this again, it looks like we may be able to use retry decorator instead of the code black context manager.  that would seemingly be cleaner but i'll take a look